docs/source/guides/error_handling.rst

```diff
@@ -0,0 +1,118 @@
+
+- `JsonError <https://github.com/cerner/beadledom/blob/master/json-common/src/main/java/com.cerner.beadledom.json.common/model/JsonError.java>`_
+  - A class that can be used to build a JSON response.
+- `ErrorDetail <https://github.com/cerner/beadledom/blob/master/json-common/src/main/java/com.cerner.beadledom.json.common/model/ErrorDetail.java>`_
+  - A class that can be used to add additional error details to the JSON response.
+
+These models are serializable objects that can be set as the response's entity body.
+
+Example Response:
+
+.. code-block:: json
+
+  {
+    "errors" [
+      {
+        "domain": "global",
+        "reason": "notFound",
+        "message": "Not Found"
+      }
+    ],
+    "code": 404,
+    "message": "Not Found"
+  }
+
+Example Response (without additional errors information):
+
+.. code-block:: json
+
+  {
+    "code": 404,
+    "message": "Not Found"
+  }
+
+
+Behavior
+________
+
+Without the use of an `ExceptionMapper <http://docs.oracle.com/javaee/6/api/javax/ws/rs/ext/ExceptionMapper.html>`_,
+an unhandled exception thrown from a resource endpoint will return an HTTP response in plain text
+or HTML that may contain stack traces or other system-specific information that should not be displayed to the end-user.
+
+Beadledom addresses this concern by implementing exception mappers. By default, Beadledom comes
+packaged with three exception mappers that handle this functionality behind the scenes:
+
+- `WebApplicationExceptionMapper <https://github.com/cerner/beadledom/blob/master/jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmapping/WebApplicationExceptionMapper.java>`_
+  - An exception mapper for exceptions in the WebApplicationException family.
+- `FailureExceptionMapper <https://github.com/cerner/beadledom/blob/master/resteasy/src/main/java/com/cerner/beadledom/resteasy/exceptionmapping/FailureExceptionMapper.java>`_
+  - An exception mapper for exceptions in the Failure family, including Resteasy exceptions.
+- `ThrowableExceptionMapper <https://github.com/cerner/beadledom/blob/master/jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmapping/ThrowableExceptionMapper.java>`_
+  - A catch-all exception mapper for all other unhandled exceptions.
+
+Each of these ExceptionMapper classes has a distinct role in a RESTful service.
+
+The WebApplicationExceptionMapper is intended to handle all exceptions thrown from a service
+that extend from the WebApplicationException hierarchy. This includes exceptions such as
+BadRequestException, NotFoundException, custom exceptions extending from WebApplicationException,
+etc. WebApplicationExceptionMapper treats the exception as-is but sanitizes the exception's
+response, status, and message and formats it as JSON before sending back a response. The status
+will be propagated through to the response as long as it is a valid code; if the code is invalid,
+this will be treated as an Internal Server Error (500). The response message will be derived from
```

Comment:
nit: can we say 500 status code instead of 500?

Reply:
https://github.com/cerner/beadledom/pull/45/commits/2530c00610b01c4648943d907f7d06846a37a0a4#diff-2275a5c50f335dc3eeff64d39ae09f45R80